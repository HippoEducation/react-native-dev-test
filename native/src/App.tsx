import React from 'react';
import Posts from './components/PostView';
import { StyleSheet, Text, View, SafeAreaView, Platform } from 'react-native';

export function App() {
  const headerContainerStyle = () => Platform.OS === 'ios'
  ? osStyles.iosHeaderContainer : osStyles.androidHeaderContainer;
  const headerStyle = () => Platform.OS === 'ios'
  ? osStyles.iosHeader : osStyles.androidHeader;

  return (
    <SafeAreaView style={styles.container}>
      <View style={headerContainerStyle()}>
        <Text style={headerStyle()}>Latin Daily Reader</Text>
      </View>
      <Posts />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  headerContainer: {
    backgroundColor: '#A5FFD6',
    alignItems: 'center',
  },
  header: {
    fontSize: 20,
    marginLeft: 10,
    color: '#FF0054',
    alignContent: 'center',
    fontWeight: 'bold',
  },
});

const osStyles = StyleSheet.create({
  androidHeaderContainer: {
    ...styles.headerContainer,
    height: '12%',
  },
  iosHeaderContainer: {
    ...styles.headerContainer,
    height: '8%',
  },
  iosHeader:  {
    ...styles.header,
    marginTop: 20,
    fontFamily: 'Chalkduster',
  },
  androidHeader: {
    ...styles.header,
    marginTop: 40,
    fontFamily: 'monospace',
  }
})
