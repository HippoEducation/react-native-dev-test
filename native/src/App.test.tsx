import React from 'react';
import renderer, { act, create } from 'react-test-renderer';

import { App } from './App';
import PostItem from './components/postItem';
import PostList from './components/postList';

jest.mock();

const mockPostData = [
  {
    title: "Harry Potter and the Sorcerer's Stone",
    author: {
      id: '123',
      name: 'JK Rowling',
    },
    body: '# Three best friends taking on the world',
    published: new Date(),
    onPress: () => {},
  },
  {
    title: 'Harry Potter and the Chamber of Secrets',
    author: {
      id: '122',
      name: 'JK Rowling',
    },
    body: '# Three best friends taking on the world',
    published: new Date(),
    onPress: () => {},
  },
];

describe('<App />', () => {
  test('renders without crashing', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).not.toBeNull();
  });
});

describe('<PostComponent />', () => {
  test('check to make sure nothing extra is rendered in PostItem component', () => {
    let root;
    act(() => {
      root = create(
        <PostItem
          title={mockPostData[0].title}
          author={mockPostData[0].author}
          body={mockPostData[0].body}
          publishedDate={mockPostData[0].published}
          onPress={mockPostData[0].onPress}
        />
      );
    });

    expect(root.toJSON()).toMatchSnapshot();
  });
});

describe('<PostList />', () => {
  test('renders correctly!', () => {
    const tree = renderer.create(<PostList />).toJSON();
    expect(tree).toMatchInlineSnapshot(`
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "width": "100%",
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    `);
  });
});
