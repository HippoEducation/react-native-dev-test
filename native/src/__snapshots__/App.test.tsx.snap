// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostComponent /> check to make sure nothing extra is rendered in PostItem component 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "flex": 1,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 25,
    }
  }
>
  <Text
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
      }
    }
  >
    Harry Potter and the Sorcerer's Stone
  </Text>
  <Text
    onPress={[Function]}
    style={
      Object {
        "bottom": 0,
        "fontSize": 15,
      }
    }
  >
    By: JK Rowling
  </Text>
  <Text
    style={Object {}}
  >
    Summary: #
  </Text>
  <Text
    style={Object {}}
  >
    Published: 08 Jun 2020
  </Text>
</View>
`;
